File: AD-HOC/2247/main.js
let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
for (let n, test = 1, difference = 0; n = parseInt(lines.shift()); test++, difference = 0, console.log()) {
    console.log(`Teste ${test}`);
    while (n--) {
        let [j, z] = lines.shift().split(' ').map(Number);
        difference += j - z;
        console.log(difference);
    }
}

File: AD-HOC/2247/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        for (int n, test = 1, difference = 0; (n = scanner.nextInt()) > 0; test++, difference = 0, System.out.println()) {
            System.out.printf("Teste %d\n", test);
            while (n-- > 0) {
                int j = scanner.nextInt();
                int z = scanner.nextInt();
                difference += j - z;
                System.out.println(difference);
            }
        }
        scanner.close();
    }
}
